internal/workflow/preprocessing.go

```diff
@@ -162,3 +173,84 @@ func withLocalActOpts(ctx temporalsdk_workflow.Context) temporalsdk_workflow.Con
 		},
 	)
 }
+
+func writePREMISFile(ctx temporalsdk_workflow.Context, sipPath string) error {
+	var e error
+	metadataPath := filepath.Join(sipPath, "metadata")
+	premisFilePath := filepath.Join(metadataPath, "premis.xml")
+
+	// Add metadata directory if it doesn't exist.
+	if _, err := os.Stat(metadataPath); err != nil {
+		err = os.MkdirAll(metadataPath, 0o750)
+		if err != nil {
+			return err
+		}
+	}
+
+	// Add PREMIS objects.
+	var addPREMISObjects activities.AddPREMISObjectsResult
+	e = temporalsdk_workflow.ExecuteActivity(
+		withLocalActOpts(ctx),
+		activities.AddPREMISObjectsName,
+		&activities.AddPREMISObjectsParams{
+			SIPPath:        sipPath,
+			PREMISFilePath: premisFilePath,
+		},
+	).Get(ctx, &addPREMISObjects)
+	if e != nil {
+		return e
+	}
+
+	// Add PREMIS event noting validate SIP file formats result.
+	validateStructureOutcomeDetail := "It's good"
```

Comment:
As this very long named variable is only used once I think it's better to just use the value (which is shorter than the var name) directly in `AddPREMISEventParams()`. :laughing: